### ISSUE TITLE
tobs installer seems to work correctly, but spits out error message `Error: could not install The Observability Stack: resource name may not be empty`

### ISSUE THREAD

**Issue by alexpdp7** [timescale/tobs#296]
Hi,

I'm testing tobs on a test cluster (kind + podman).

```
$ tobs install --tracing -n tobs

WARNING: Using a generated self-signed certificate for TLS access to TimescaleDB.
         This should only be used for development and demonstration purposes.
         To use a signed certificate, use the "--tls-timescaledb-cert" and "--tls-timescaledb-key"
         flags when issuing the tobs install command.

Creating TimescaleDB tobs-certificate secret
Creating TimescaleDB tobs-credentials secret
skipping to create TimescaleDB s3 backup secret as backup option is disabled.
Couldn't find the cert-manager. The cert-manager is required to deploy OpenTelemetry. Do you want to install the cert-manager?
confirm the action by typing y or yes and press enter: y
Waiting on pod cert-manager-cainjector-8c7796555-wc6c7...
Pod cert-manager-cainjector-8c7796555-wc6c7 has started
Waiting on pod cert-manager-cdc85d4c4-wwn9x...
Pod cert-manager-cdc85d4c4-wwn9x has started
Waiting on pod cert-manager-webhook-55f45df998-xczm5...
Pod cert-manager-webhook-55f45df998-xczm5 has started
Successfully created cert-manager
Installing The Observability Stack
2022/03/31 16:34:08 release installed successfully: tobs/tobs-0.9.0
Waiting for helm install to complete...
Waiting for pods to initialize...
Waiting on pod tobs-kube-prometheus-operator-5f78bdd5c9-x5kwt...
Pod tobs-kube-prometheus-operator-5f78bdd5c9-x5kwt has started
Waiting on pod tobs-kube-state-metrics-765c569cbc-fkdqd...
Pod tobs-kube-state-metrics-765c569cbc-fkdqd has started
Waiting on pod tobs-prometheus-node-exporter-xcs6k...
Pod tobs-prometheus-node-exporter-xcs6k has started
Waiting on pod tobs-promscale-65bcd64969-fh7ns...
Pod tobs-promscale-65bcd64969-fh7ns has started
Waiting on pod tobs-timescaledb-0...
Pod tobs-timescaledb-0 has started
Waiting on pod tobs-grafana-79785ddbdb-lljj8...
Pod tobs-grafana-79785ddbdb-lljj8 has started
Waiting on pod tobs-kube-prometheus-operator-5f78bdd5c9-x5kwt...
Pod tobs-kube-prometheus-operator-5f78bdd5c9-x5kwt has started
Waiting on pod tobs-kube-state-metrics-765c569cbc-fkdqd...
Pod tobs-kube-state-metrics-765c569cbc-fkdqd has started
Waiting on pod tobs-promscale-65bcd64969-fh7ns...
Pod tobs-promscale-65bcd64969-fh7ns has started
Waiting on pod tobs-grafana-db-4szgs...
Pod tobs-grafana-db-4szgs has started
Waiting on pod ...
Error: could not install The Observability Stack: resource name may not be empty
```

Despite the error message, deployments look fine:

```
$ kubectl get all -n tobs
NAME                                                             READY   STATUS      RESTARTS      AGE
pod/alertmanager-tobs-kube-prometheus-alertmanager-0             2/2     Running     0             46m
pod/opentelemetry-operator-controller-manager-68b8cf9bb9-hbq8v   2/2     Running     0             46m
pod/prometheus-tobs-kube-prometheus-prometheus-0                 2/2     Running     0             46m
pod/tobs-grafana-79785ddbdb-lljj8                                3/3     Running     3 (44m ago)   46m
pod/tobs-grafana-db-4szgs                                        0/1     Completed   2             46m
pod/tobs-jaeger-59f9b94f7f-4vxpb                                 1/1     Running     0             46m
pod/tobs-kube-prometheus-operator-5f78bdd5c9-x5kwt               1/1     Running     0             46m
pod/tobs-kube-state-metrics-765c569cbc-fkdqd                     1/1     Running     0             46m
pod/tobs-prometheus-node-exporter-xcs6k                          1/1     Running     0             46m
pod/tobs-promlens-7685c9744d-h7gwq                               1/1     Running     0             46m
pod/tobs-promscale-65bcd64969-fh7ns                              1/1     Running     5 (45m ago)   46m
pod/tobs-timescaledb-0                                           1/1     Running     0             46m

NAME                                                                TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                      AGE
service/alertmanager-operated                                       ClusterIP   None            <none>        9093/TCP,9094/TCP,9094/UDP   46m
service/opentelemetry-operator-controller-manager-metrics-service   ClusterIP   10.96.68.185    <none>        8443/TCP                     46m
service/opentelemetry-operator-webhook-service                      ClusterIP   10.96.178.68    <none>        443/TCP                      46m
service/prometheus-operated                                         ClusterIP   None            <none>        9090/TCP                     46m
service/tobs                                                        ClusterIP   10.96.47.123    <none>        5432/TCP                     46m
service/tobs-config                                                 ClusterIP   None            <none>        8008/TCP                     46m
service/tobs-grafana                                                ClusterIP   10.96.18.37     <none>        80/TCP                       46m
service/tobs-jaeger                                                 ClusterIP   10.96.136.17    <none>        16686/TCP                    46m
service/tobs-kube-prometheus-alertmanager                           ClusterIP   10.96.177.236   <none>        9093/TCP                     46m
service/tobs-kube-prometheus-operator                               ClusterIP   10.96.147.6     <none>        443/TCP                      46m
service/tobs-kube-prometheus-prometheus                             ClusterIP   10.96.165.82    <none>        9090/TCP                     46m
service/tobs-kube-state-metrics                                     ClusterIP   10.96.152.37    <none>        8080/TCP                     46m
service/tobs-prometheus-node-exporter                               ClusterIP   10.96.117.75    <none>        9100/TCP                     46m
service/tobs-promlens                                               ClusterIP   10.96.196.242   <none>        80/TCP                       46m
service/tobs-promscale-connector                                    ClusterIP   10.96.135.163   <none>        9201/TCP,9202/TCP            46m
service/tobs-replica                                                ClusterIP   10.96.45.126    <none>        5432/TCP                     46m

NAME                                           DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
daemonset.apps/tobs-prometheus-node-exporter   1         1         1       1            1           <none>          46m

NAME                                                        READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/opentelemetry-operator-controller-manager   1/1     1            1           46m
deployment.apps/tobs-grafana                                1/1     1            1           46m
deployment.apps/tobs-jaeger                                 1/1     1            1           46m
deployment.apps/tobs-kube-prometheus-operator               1/1     1            1           46m
deployment.apps/tobs-kube-state-metrics                     1/1     1            1           46m
deployment.apps/tobs-promlens                               1/1     1            1           46m
deployment.apps/tobs-promscale                              1/1     1            1           46m

NAME                                                                   DESIRED   CURRENT   READY   AGE
replicaset.apps/opentelemetry-operator-controller-manager-68b8cf9bb9   1         1         1       46m
replicaset.apps/tobs-grafana-79785ddbdb                                1         1         1       46m
replicaset.apps/tobs-jaeger-59f9b94f7f                                 1         1         1       46m
replicaset.apps/tobs-kube-prometheus-operator-5f78bdd5c9               1         1         1       46m
replicaset.apps/tobs-kube-state-metrics-765c569cbc                     1         1         1       46m
replicaset.apps/tobs-promlens-7685c9744d                               1         1         1       46m
replicaset.apps/tobs-promscale-65bcd64969                              1         1         1       46m

NAME                                                              READY   AGE
statefulset.apps/alertmanager-tobs-kube-prometheus-alertmanager   1/1     46m
statefulset.apps/prometheus-tobs-kube-prometheus-prometheus       1/1     46m
statefulset.apps/tobs-timescaledb                                 1/1     46m

NAME                        COMPLETIONS   DURATION   AGE
job.batch/tobs-grafana-db   1/1           2m52s      46m
```

**Comment by ZQiannnn** [comment]
I think this func get a null podName
```
func (c *clientImpl) KubeWaitOnPod(namespace string, podName string) error {
	fmt.Printf("Waiting on pod %v...\n", podName)
	for i := 0; i < 6000; i++ {
		pod, err := c.CoreV1().Pods(namespace).Get(context.Background(), podName, metav1.GetOptions{})
		if err != nil {
			return err
		}

		podStatus := pod.Status.Phase
		if podStatus != corev1.PodPending && podStatus != corev1.PodFailed && podStatus != corev1.PodUnknown {
			fmt.Printf("Pod %v has started\n", podName)
			break
		} else if i == 5999 {
			fmt.Println("WARNING: pod did not come up in 10 minutes")
		}
		time.Sleep(100 * time.Millisecond)
	}

	return nil
}
```

**Comment by paulfantom** [comment]
@alexpdp7 are you by any chance using Mac with M1 processor?

**Comment by jamessewell** [comment]
I get this behaviour on Mac M1 -> KIND on Docker and also Amazon Linux -> EKS

**Comment by alexpdp7** [comment]
At least one thing is broken- I think- due to this. All the docs seem to imply an OpenTelemetry collector will be automatically installed (although I couldn't find it in the source code easily), and I don't get that- I get the operator, but not custom resource or anything.

**Comment by paulfantom** [comment]
@alexpdp7 you are correct, CR should be installed, but it isn't when using helm. Both of those issues are fixed in https://github.com/timescale/tobs/pull/305 and https://github.com/timescale/tobs/pull/314. Sorry for the long wait in resolving this problems.

**Comment by Rajakavitha1** [comment]
the same error appears but pods are in crash loopback status.
```
rajakavithakodhandapani@Rajakavithas-MacBook-Pro-2 ~ % kubectl get pods
NAME                                                         READY   STATUS             RESTARTS       AGE
alertmanager-tobs-kube-prometheus-alertmanager-0             2/2     Running            0              9m50s
opentelemetry-operator-controller-manager-68b8cf9bb9-5pmpp   2/2     Running            0              9m55s
prometheus-tobs-kube-prometheus-prometheus-0                 0/2     Pending            0              9m52s
tobs-grafana-7c74fd4cd7-z6k27                                2/3     CrashLoopBackOff   6 (4m6s ago)   9m55s
tobs-jaeger-6c9848774-g4qdg                                  1/1     Running            0              9m55s
tobs-kube-prometheus-operator-5f78bdd5c9-fqzdn               1/1     Running            0              9m55s
tobs-kube-state-metrics-765c569cbc-w57kd                     1/1     Running            0              9m51s
tobs-prometheus-node-exporter-4x6mq                          1/1     Running            0              9m55s
tobs-prometheus-node-exporter-5gn2g                          1/1     Running            0              9m55s
tobs-prometheus-node-exporter-q8cbm                          1/1     Running            0              9m55s
tobs-promlens-7685c9744d-4m6sd                               1/1     Running            0              9m55s
tobs-promscale-64fd4d9ccb-svwwr                              1/1     Running            0              9m48s
```

**Comment by paulfantom** [comment]
@Rajakavitha1 what is the tobs version that you used and what was you method of installation?

Bear in mind that the fix to the issue listed here should be generally available when tobs 0.10.0 lands (we are in the middle of a release process).

**Comment by paulfantom** [comment]
We released 0.10.0 and I cannot replicate this bug in that release. As such I am closing this PR. If you see it happening, please reopen or comment.

**Comment by alexpdp7** [comment]
(Sorry, I tried to reproduce with the new release, but I have different issues- and they maybe lie with my cluster. Will continue investigating and raise an issue if necessary.)
